### ISSUE TITLE
[Feature request] `__LABEL__`

### ISSUE THREAD

**Issue by aaaaaa123456789** [gbdev/rgbds#1158]
The idea is simple here: an RGBASM-defined `EQUS` that expands to the fully-qualified form of the last label declared so far in the current section. If there's no such label, it should be undefined, raising an error when used.

```
SECTION "Test", WRAM
; __LABEL__ is undefined here
wFoo:
  print "{__LABEL__}" ; prints wFoo
.bar
  print "{__LABEL__}" ; prints wFoo.bar
wBaz:
{__LABEL__}.quux: ; declares wBaz.quux
```

This could be used both for code-generation purposes and for macros that need to refer to the current address in some way. (I believe @ISSOtm would have an immediate use for `__LABEL__ + (@ - __LABEL__)` in his debugfile generation macros.)

**Comment by ISSOtm** [comment]
Duplicate of #775, like #1156... :P

**Comment by aaaaaa123456789** [comment]
> Duplicate of #775, like #1156... :P

#775 is about parent scopes, so it's not the same. It would also most likely answer the concern about multi-level scopes, if that comes up, since you can just parse `__LABEL__` for whatever level you want.

**Comment by zlago** [comment]
is
```asm
Global:
..local:
```
valid rgbds?

**Comment by aaaaaa123456789** [comment]
No, but there's a request/proposal to make it happen someday, hence the request to not block that.

**Comment by zlago** [comment]
so, mapping the number of `.`s to the scope could maybe be implemented?
(..is what im saying even relevant? ><')
```asm
; example of the concept, in case its not clear
l1:
.l2: ; equivalent to l1.l2:
..l3: ; equivalent to l1.l2.l3:
```
